### DISCUSSION TITLE
To pass a variable as a value of the '--template' option

### DISCUSSION THREAD

**Discussion by johnmapeson** [jgm/pandoc#9790] [Q&A] [answered]
I have an HTML template stored as a string variable, that is, `echo $my_custom_html5_template` returns

```
<!DOCTYPE html>
etc.
```

Is it possible to pass this variable as a value of the `--template` option?

That is, instead of using 

```
pandoc input.md -s --template="my_custom_html5_template.html5" -o output.htm
```

I want to use something like

```
pandoc input.md -s --template=$my_custom_html5_template -o output.htm
```

Is it possible somehow?

**Comment by jgm** [comment]
Using zsh "process substitution"
```
pandoc input.md -s --template =(echo $my_custom_html5_template) -o output.htm
```
almost works, but not quite, because pandoc looks for an appropriate extension on the template filename and if it isn't found, it tries to find it in the `templates` directory of the user data directory.  Removing this constraint might allow it to work.

**Reply by jgm** [reply, marked as answer]
Of course you could always use `mktemp` and create a temporary file to hold the template.

**Reply by johnmapeson** [reply]
Many thanks, John!

**Reply by johnmapeson** [reply]
John, sorry for a late reply.

```
my_custom_html5_template=$(grep -v '<title>' <<< $(pandoc -D html))
pandoc input.md -s --template =(echo $my_custom_html5_template) -o output.htm
```

The first line work fine, but the second returns

```
Could not find data file templates/zshQ2i6xH.html
```

where `zshQ2i6xH` is a random string.

If I understand you correctly, you said that the second line doesn't work in its current state because

> pandoc looks for an appropriate extension on the template filename and if it isn't found, it tries to find it in the templates directory of the user data directory.

But what I see from the error message above is that extension seems to be guessed correctly. Nevertheless, the line itself doesn't work. What might be a reason?

**Reply by cagix** [reply]
@jsx97 `--template` expects a *file name* and searches for this file. in your example, however, you are passing a _text_, which is then interpreted as a file name.

@jgm it would be nice if there was an option to directly pass a template as text. this would avoid the need to handle temporary files. ...

**Reply by johnmapeson** [reply]
@cagix Could you show how to use a temporary file here?

**Reply by cagix** [reply]
@jsx97 hmmm, i'd use something like

```
TMPFILE=`mktemp`
pandoc -D latex >> $TMPFILE
...
pandoc --template=($TMPFILE) ...
```

(not tested!)
